src/shared_modules/content_manager/src/action.hpp

```diff
@@ -197,6 +219,7 @@ class Action final
     std::thread m_schedulerThread;
     std::atomic<bool> m_schedulerRunning = false;
     std::atomic<bool> m_actionInProgress;
+    const std::atomic<bool>* m_shouldRun;
```

Comment:
Improve (design): Don't make use of raw pointers

Severe (design): This implementation is brittle, we are passing a pointer between classes